components/org.wso2.carbon.um.ws.service/src/main/java/org/wso2/carbon/um/ws/service/ClaimManagerService.java

```diff
@@ -99,7 +99,12 @@ public String getAttributeNameFromDomain(String domainName, String claimURI) thr
 
     public Claim getClaim(String claimURI) throws UserStoreException {
         try {
-            return  getClaimManager().getClaim(claimURI);
+            org.wso2.carbon.user.api.Claim apiClaim = getClaimManager().getClaim(claimURI);
+            if (apiClaim instanceof Claim) {
+                return (Claim) apiClaim;
+            }
+            throw new UserStoreException("Returned Claim is an instance of " + apiClaim.getClass().getName()
+                    + " but an instance of org.wso2.carbon.user.core.claim.Claim is expected" );
```

Comment:
Can you change this to something like "Returned Claim object is of class ...... . Object of either org.wso2.carbon.user.core.claim.Claim class or its subclass is Expected"